### ISSUE TITLE
Is it possible to inject custom html into body?

### ISSUE THREAD

**Issue by bebraw** [ampedandwired/html-webpack-plugin#11]
Currently I'm injecting some Markdown to my page through React. It gets converted to HTML first by a loader, after that React takes care of it. This doesn't seem particularly nice way to deal with it, though.

Is it possible to inject the custom HTML generated by my Markdown loader within body somehow?

**Comment by jantimon** [comment]
#15 allows to pick up any HTML file and adds the necessary script tags to it.

Would this solve this issue?

**Comment by bebraw** [comment]
@jantimon This could actually work. I guess it would be just a matter of putting a `require` within HTML and letting Webpack expand it`

**Comment by kairyou** [comment]
Hope to add require() support, It looks great:
new HtmlWebpackPlugin({
	filename: 'index.html',
	template: 'src/_index.html',
})
src/_index.html:
```html
<!DOCTYPE html>
<html>
<head>
</head>
<body>
{% require('./mod/header.html') %}
{% require('./mod/main.html') %}
{% require('./mod/sub.html') %}
{% require('./mod/footer.html') %}
</body>
</html>
```

**Comment by jantimon** [comment]
@kairyou Webpack turns `require` calls into runtime code.
Unfortunately the index.html is precompiled so it looks like this is not possible.

**Comment by bebraw** [comment]
@jantimon Maybe we need to do this in two passes? The first pass would work outside of normal Webpack context (just triggers loader?) and expand the injected parts. It would generate the template used by Webpack in the second pass. I have a feeling this could be quite powerful if we can make it work.

**Comment by ampedandwired** [comment]
Possibly could be accomplished by using a loader in combination with this plugin, I think #10 is relevant.

**Comment by bebraw** [comment]
I see a lot is going on with the project. Props for that!

Do you think it might be possible to load Markdown into HTML somehow now? Anything I can do to help?

**Comment by SimenB** [comment]
@bebraw I'm able to inject HTML from markdown using my PR #36

**Comment by jantimon** [comment]
The `templateContent` option allows you to use a function (which might process markdown).

**Comment by jantimon** [comment]
I'll close this for now as it is already possible with `templateContent` and might be solved if #10 is implemented.
